test/utilities.go

```diff
@@ -0,0 +1,196 @@
+	}
+
+	// This func uses jsondiff.
+	// options configures it.
+	options := jsondiff.Options{
+		Added:            jsondiff.Tag{Begin: "___ADDED___", End: ""},
+		Removed:          jsondiff.Tag{Begin: "___REMED___", End: ""},
+		Changed:          jsondiff.Tag{Begin: "___DIFFER___", End: ""},
+		ChangedSeparator: " -> ",
+	}
+	options.PrintTypes = false
+	d, str := jsondiff.Compare([]byte(j1), []byte(j2), &options)
+	// If fully match, return true
+	if d == jsondiff.FullMatch {
+		return true, nil
+	}
+
+	// Check the difference, and decide if it is still accepted as "equal"
+	// Scan line by line, and examine the difference
+	scanner := bufio.NewScanner(strings.NewReader(str))
+	for scanner.Scan() {
+		line := scanner.Text()
+		// If baseline has the property (j1), but the test (j2) has it removed
+		if strings.Contains(line, "___REMED___") {
+			// If the value of the property is false, then accept it
+			// The default value is false. Missing property is considered as false
+			propValue := strings.SplitN(line, ": ", 2)
+			if len(propValue) == 1 {
+				addedLine := strings.SplitN(propValue[0], "___REMED___", 2)
+				if len(addedLine) == 2 && (addedLine[1] == "}," || addedLine[1] == "}") {
+					continue
+				}
+			}
+			if len(propValue) == 2 &&
+				(propValue[1] == "null," ||
+					propValue[1] == "[]," || propValue[1] == "[]" ||
+					propValue[1] == "false," || propValue[1] == "false" ||
+					propValue[1] == "0," || propValue[1] == "0" ||
+					propValue[1] == "\"\"," || propValue[1] == "\"\"" ||
+					propValue[1] == "{") {
+				continue
+			}
+			// Any other ommision will not be considered as equal
+			err = fmt.Errorf(line)
+			return false, err
+		}
+		// If the test has properties not found in the baseline, then they are not equal
+		if strings.Contains(line, "___ADDED___") {
+			//			fmt.Printf("XXX j1:\n%s\n", j1)
+			//			fmt.Printf("XXX j2:\n%s\n", j2)
+			err = fmt.Errorf(line)
+			return false, err
+		}
+		// If the properties are different, they they are not equal
+		if strings.Contains(line, "___DIFFER___") {
+			err = fmt.Errorf(line)
+			return false, err
+		}
+	}
+	// Extra layer of check. This is not expected to be true.
+	// If only difference is removed with false value property, then the difference
+	// will be qualified as SupersetMatch. If not, should be be considered equal.
+	if d != jsondiff.SupersetMatch {
+		err = fmt.Errorf(str)
+		return false, err
+	}
+	return true, nil
+}
+
+// SortJsonArrays sorts the elements inside json arrays.
+// Some new line organization may be broken.
+// e.g. ...[  {a: "a3", b: "b3"},   {a: "a1", b: "b1"},    {a: "a2", b: "b2"}]...
+// will be:
+//      ...[{a: "a1", b: "b1"},{a: "a2", b: "b2"},{a: "a3", b: "b3"}]...
+// Note: any characters between [ and {, as well as between elements will be lost: ("...},xxx{..." "xxx" is lost ) 
+// This function will recursively sort arrays inside arrays as well.
+func SortJsonArrays(json string) (sorted string, err error) {
```

Comment:
Do the tests pass if you only sort the maps `{}`, and not the arrays `[]`?

Reply:
I only sort the arrays.